### PR TITLE
v2.0, breaking change!! New decoded config, ability to handle new levels

### PR DESCRIPTION
**Pull request by Andy1210**

### CODE REVIEW

File: Peggle/Peggle.toc [hunk 1]
```diff
@@ -1,9 +1,10 @@
-## Interface: 80501
+## Interface: 80100
```
Comment (Andy1210) [inline Peggle/Peggle.toc:1]: note: 80100 is the current valid Interface version
image

File: Peggle/Peggle.lua [hunk 2]
```diff
@@ -902,74 +920,58 @@ local function L(e, t)
...
 	t = A(c(e, 4, 4));
 	o = A(c(e, 3, 3));
 	l = A(c(e, 2, 2));
+
 	if((i == h)and(a == u)and(t == f)and(o == d)and(l == r))then
 		return n;
 	end
 end
 local function p(t)
+	t = string.match(t, "(.*)%-(.*)") or t
```
Comment (Andy1210) [inline Peggle/Peggle.lua:923]: cutting down realm name here to fix duel/battle

File: Peggle/Peggle.lua [hunk 3]
```diff
@@ -10419,6 +10397,7 @@ local function k()
 		end
 	end);
 	i:SetScript("OnEvent", function(m, n, C, i, l, a)
+		a = string.match(a, "(.*)%-(.*)") or a
```
Comment (Andy1210) [inline Peggle/Peggle.lua:10388]: cutting down realm name here to fix duel/battle

File: Peggle/Peggle.lua [hunk 4]
```diff
@@ -69,11 +74,10 @@ local t = function(...)
...
-};
 e.GetBackdrop = function()
 	return{bgFile = "Interface\\Tooltips\\UI-Tooltip-Background", tileSize = 16, edgeFile = "Interface\\Tooltips\\UI-Tooltip-Border", tile = 1, edgeSize = 16, insets = {top = 5, right = 5, left = 5, bottom = 5, }}
 end
+C_ChatInfo.RegisterAddonMessagePrefix(e.addonName);
```
Comment (Andy1210) [inline Peggle/Peggle.lua:81]: Register Addon prefix to allow it to send messages on the hidden addon channel (to fix duel/battle)

**Comment by adamz01h** [comment]
Thanks for all the work on this!

**Pull request merged**